echoregions/regions2d/regions2d.py

```diff
@@ -564,30 +571,89 @@ def transect_mask(
         ----------
         da_Sv : Data Array
             DataArray of shape (ping_time, depth) containing Sv data.
-        transect_dict : dict
-            Dictionary for transect values. Values must be unique.
+        transect_sequence_type_dict : dict
+            Dictionary for transect sequence type values. The values denote where in the context
+            of the transect each region lays in, i.e. whether we are at the beginning of the
+            transect, a break in the transect, a resumption of the transect, or at the end
+            of the transect.
+        transect_sequence_type_next_allowable_dict : dict
+            Dictionary for the allowable transect sequence type value(s) that can follow a
+            transect sequence type value.
         bbox_distance_threshold: float
-            The maximum value for how far apart the left and right bounding box for each transect
-            value region. Default is set to 1 minute.
+            Maximum allowable value between the left and right bounding box timestamps
+            for each transect value region. Default is set to 1 minute.
+        must_pass_check : bool
+            True: Will check transect strings to enforce sequence rules. If this check
+            encounters any incorrect transect type sequence orders or wider than bbox distance
+            threshold regions, it will raise an exception.
+            False: Will still check transect strings but will instead just print out warnings
+            for violations of the above mentioned sequence rules.
 
         Returns
         -------
         M : Data Array
-            A DataArray masked by the transect values from the Regions2d.data dataframe
-            with dimensions (ping_time, depth).
+            A binary DataArray with dimensions (ping_time, depth) where 1s are within transect
+            and 0s are outside transect.
         """
 
+        def _check_transect_sequences(transect_df, must_pass_check):
```

Comment:
Seems that this could be moved outside of `transect_mask`?

Reply:
Done!